x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/ExplainLifecycleIT.java

```diff
@@ -0,0 +1,206 @@
+            assertThat(onlyErrorsResponse, allOf(not(hasKey(goodIndex)), not(hasKey(unmanagedIndex))));
+        });
+    }
+
+    public void testExplainIndexContainsAutomaticRetriesInformation() throws Exception {
+        createFullPolicy(client(), policy, TimeValue.ZERO);
+
+        // create index without alias so the rollover action fails and is retried
+        createIndexWithSettingsNoAlias(client(), index, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(LifecycleSettings.LIFECYCLE_NAME, policy)
+        );
+
+        assertBusy(() -> {
+            Map<String, Object> explainIndex = explainIndex(client(), index);
+            assertThat((Integer) explainIndex.get(FAILED_STEP_RETRY_COUNT_FIELD), greaterThanOrEqualTo(1));
+            assertThat(explainIndex.get(IS_AUTO_RETRYABLE_ERROR_FIELD), is(true));
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testExplainIndicesWildcard() throws Exception {
+        createNewSingletonPolicy(client(), policy, "delete", new DeleteAction(), TimeValue.timeValueDays(100));
+        String firstIndex = this.index + "-first";
+        String secondIndex = this.index + "-second";
+        String unmanagedIndex = this.index + "-unmanaged";
+        String indexWithMissingPolicy = this.index + "-missing_policy";
+        createIndexWithSettings(client(), firstIndex, alias + firstIndex, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(LifecycleSettings.LIFECYCLE_NAME, policy));
+        createIndexWithSettings(client(), secondIndex, alias + secondIndex, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(LifecycleSettings.LIFECYCLE_NAME, policy));
+        createIndexWithSettings(client(), unmanagedIndex, alias + unmanagedIndex, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0));
+        String missingPolicyName = "missing_policy_";
+        createIndexWithSettings(client(), indexWithMissingPolicy, alias + indexWithMissingPolicy, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(LifecycleSettings.LIFECYCLE_NAME, missingPolicyName));
+
+        assertBusy(() -> {
+            Map<String, Map<String, Object>> explain = explain(client(), this.index + "*", false, false);
+            assertManagedIndex(explain.get(firstIndex));
+            assertManagedIndex(explain.get(secondIndex));
+            assertUnmanagedIndex(explain.get(unmanagedIndex));
+
+            Map<String, Object> explainIndexWithMissingPolicy = explain.get(indexWithMissingPolicy);
```

Comment:
I think we could probably put this one is a second `assertBusy` so that we don't end up doing two explains when we only need to do one

Reply:
There's only one explain API call in this assertBusy. I believe it's ok to leave it in one `assertBusy` call.